pkgs/applications/version-management/pijul/default.nix

```diff
@@ -4,21 +4,32 @@ with rustPlatform;
 
 buildRustPackage rec {
   name = "pijul-${version}";
-  version = "0.4.4";
+  version = "0.6.0";
 
-  src = fetchurl {
+  raw_src = fetchurl {
     url = "https://pijul.org/releases/${name}.tar.gz";
-    sha256 = "8f133b7e14bfa84156c103126d53b12c6dfb996dcdebcf1091199ff9c77f3713";
+    sha256 = "a6b066b49b25d1083320c5ab23941deee795e1fcbe1faa951e95189fd594cdb3";
   };
 
+  src = "${name}.tar.gz";
+
   sourceRoot = "${name}/pijul";
 
+  preUnpack = ''
+     tar -cf empty.tar -T /dev/null
+     cp ${raw_src} ${name}.tar.gz
+     chmod u+w ${name}.tar.gz
+     tar -xf ${name}.tar.gz --transform=s,^,${name}/,
+     tar -cf ${name}.tar.gz ${name}
+     rm -rf ${name}
+  '';
```

Comment:
Did you try just `sourceRoot = ".";` instead of repacking?

I do agree that it would be nice if the upstream tarball was what you obtain by repacking, though.